kubespawner/reflector.py

```diff
@@ -29,12 +28,46 @@ class PodReflector(SingletonConfigurable):
         """
     )
 
-    pods = Dict(
+    resources = Dict(
         {},
         help="""
-        Dictionary of pod names to Pod objects.
+        Dictionary of resource names to the appropriate resource objects.
+
+        This can be accessed across threads safely.
+        """
+    )
+
+    kind = Unicode(
+        'resource',
+        help="""
+        Human readable name for kind of object we're watching for.
+
+        Used for diagnostic messages.
+        """
+    )
+
+    list_method_name = Unicode(
+        None,
+        allow_none=True,
```

Comment:
From the usage below, `list_method_name` is *not* allowed to be None, as `getattr(obj, None)` will always fail. Probably best to leave this without setting None or allow_none.